go/types/subtype_test.go

```diff
@@ -262,8 +262,8 @@ func TestAssertTypeCycleUnion(tt *testing.T) {
 		StructField{"y", MakeUnionType(NumberType, StringType), false},
 	)
 
-	assert.True(tt, isSubtype(t2, t1, nil))
-	assert.False(tt, isSubtype(t1, t2, nil))
+	assert.True(tt, isSubtype(t2, t1, true, nil))
```

Comment:
Yeah, I don't like the extra params. It is too hard to reason about if you are not aware of the motivation of this weird case.